src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToRfc3339Transformer.php

```diff
@@ -69,6 +69,10 @@ public function reverseTransform($rfc3339)
             return;
         }
 
+        if (!preg_match('/^(\d{4})-(\d{2})-(\d{2})T\d{2}:\d{2}(?::\d{2})?(?:\.\d)?(?:Z|(?:(?:\+|-)\d{2}:\d{2}))$/', $rfc3339, $matches)) {
```

Comment:
In theory, if someone doesn't use their browser but submits dates manually that could indeed feel like a BC break to them. But since #28372 this transformer isn't used anymore by any of the built-in form types. So yes, maybe we should relax this pattern here and in `DateTimeToHtml5LocalDateTimeTransformer` a bit. The question then is, how much relaxation should we do and when is a failure acceptable?

Reply:
Just keeping the previous one? `'/(\d{4})-(\d{2})-(\d{2})/'`? (should be also applied to `DateTimeToHtml5LocalDateTimeTransformer`) - or at least put the added trailing pattern is a `(?:...)?` to make them optional?

Reply:
I don't understand this. If we're implementing RFC 339, we can't discuss about the regexp, right? We should use the one defined in that RFC. Since we're fixing a bug, BC breaks are not considering. Strictly speaking, all bug fixes are BC breaks because we're changing the previous behaviour.

Reply:
Javier has a valid point here. Let's just keep it the way it is.

Reply:
That is hitting users now, see #28703 
We could have done better here IMHO.
Let's learn and not reproduce...

Reply:
That's bad 😞  Shouldn't we then (later) rename this class? If we don't implement RFC 3339 as explained in https://www.ietf.org/rfc/rfc3339.txt ... that could be confusing to users. Or at least add some comment explaining why we do this? Thanks!

Reply:
I've not checked, but is usage like #28703 supposed to work? I mean, it worked before... by accident, right? Not saying we should not fix the BC break of course.

Reply:
We were accepting any parseable date, with the format allowed by the date parse of PHP. That's a working validation strategy to me also.

Reply:
The date parser accepts way too much input (see #28455 for such an example). I don't see how that is better.

Reply:
I won't argue about how being stricter can be better (or not, e.g. for accessibility) - but this is still a regression.